index.js

```diff
@@ -0,0 +1,69 @@
+const { v4: uuidv4 } = require('uuid');
+
+async function getDomainUrl(zipModifier) {
+  let domainUrl;
+  await zipModifier.modifyFiles(/index.html$/, async (content) => {
+    const pageUrlRegex = content.match(/<!-- Mirrored from (.*) by HTTrack/);
+    if (!pageUrlRegex) {
+      return content;
+    }
+    const pageUrl = pageUrlRegex[1];
+    domainUrl = pageUrl.substr(0, pageUrl.indexOf('/'));
+    return content;
+  });
+
+  return domainUrl;
+}
+
+async function addImages(zipModifier, { content, images, baseUrl } = {}) {
+  if (!images) {
+    return;
+  }
+
+  await Promise.all(
+    images
+      .filter((image) => image !== null)
+      .map(async (image) => {
+        const imageUrl = baseUrl + image.replace(/\"/g, '');
+        if (imageUrl.indexOf('http') < 0) {
+          imageUrl = 'http://' + imageUrl;
+        }
+        const body = await fetch(imageUrl);
+        const fileName = imageUrl.substr(imageUrl.lastIndexOf('/') + 1);
+        const fileExtensionRegex = /\.(jpg|png|jpeg|gif)/gi;
+        const fileExtension = fileName.match(fileExtensionRegex);
+        const uniqueFileName = uuidv4() + fileExtension;
+
+        content = content.replace(image, 'images/' + uniqueFileName);
+        return addNewFile(zipModifier, {
+          content: body.body,
+          path: 'web/images/' + uniqueFileName,
+          newTxtFields: { URL: imageUrl }
+        });
+      })
+  );
+
+  return content;
+}
+
+async function addNewFile(
+  zipModifier,
+  { content, path, newTxtFields: { localfile, URL, ...restOfFields } = {} } = {}
+) {
+  const newTxtModifier = new NewTxtModifier(zipModifier);
+  await zipModifier.addFile(path, content, '');
+  await newTxtModifier.addRecord(
+    {
+      localfile: localfile || path,
+      URL: URL + uuidv4(),
```

Comment:
That was meant to give unique names for some files in the Txt file mapping due to a problem we had with relative filenames. Now that I think about that, it will make sense only when called from within `addImages`. So maybe better to give it as an option to `addNewFile`? Something like `uniqueUrl`?

Reply:
Yes, or if we dont want to make "preparations for AC" we can just rely on the callers to generate this unique url themselves